### ISSUE TITLE
Runtime.startHats calls goToNextBlock on retired threads

### ISSUE THREAD

**Issue by apple502j** [LLK/scratch-vm#2576]
### Expected Behavior
goToNextBlock is not called for retired threads, since they don't have a stack to go to

### Actual Behavior
goToNextBlock gets called for retired threads. This is unnoticeable because of a bug in retireThread (#2574)

### Steps to Reproduce
1. Read source code of execute.js
2. Find that the thread can retire
3. Add this code to startHats forEach loop to confirm
```
            if (thread.status === Thread.STATUS_DONE) {
                // There is no "next" block; thread is retired
                console.log('Thread is retired, skipping');
                return;
            }
```
### Operating System and Browser
any

**Comment by BryceLTaylor** [comment]
@apple502j can you explain what problems this causes for a user?

**Comment by apple502j** [comment]
@BryceLTaylor Currently none, but this should be fixed when we fix #2574

**Comment by BryceLTaylor** [comment]
If we find that this causes user facing problems we can reopen this issue, but until then, I'm closing it.
